src/realm/sync/network/default_socket.cpp

```diff
@@ -507,6 +510,10 @@ void DefaultSocketProvider::start()
 
     m_logger_ptr->trace("Default event loop: start()");
     REALM_ASSERT(m_state == State::Stopped);
+
+    // Reset the service so it can be run() again
+    m_service.reset();
```

Comment:
Is this ever actually needed? Currently the default socket provider is only used once, except in testing.

Reply:
It was only for completeness if the default socket provider ever needed to be restarted. I can remove it.